### DISCUSSION TITLE
APC Pdu Not collecting data

### DISCUSSION THREAD

**Discussion by p1nn0chi0** [opendcim/openDCIM#1464] [Q&A] [answered]
I have added cdu's in the cabinet.  I have an IP address for the APC pdu's.  I have enabled a cronjob for poll_pdu_stats.php.  The oids that I have put in the template, none have made any changes to collecting data from the pdu.  The only change I can get is the status changes to up on port 1.  I did an snmp walk and was able to see the opendcim connecting to the device.  Is there something that I'm missing with the configuration for the pdu beyond the oids?  Is it possible to pull the status up or down on the individual outlets?

**Comment by timlegge** [comment]
What model of PDU?

**Reply by p1nn0chi0** [reply]
apc ap8841

**Comment by wilpig** [comment]
What is the output of this sql query?

```
SELECT a.PDUID, a.BreakerSize, b.Voltage, b.Amperage, d.SNMPVersion, b.Multiplier, b.OID1, b.OID2, b.OID3, 
			b.ProcessingProfile, b.Voltage, c.Label, c.SNMPFailureCount, e.Location, f.Name 
			FROM fac_PowerDistribution a, fac_CDUTemplate b, fac_Device c, fac_DeviceTemplate d, fac_Cabinet e, fac_DataCenter f 
			WHERE a.PDUID=c.DeviceID and a.TemplateID=b.TemplateID AND a.TemplateID=d.TemplateID and a.CabinetID=e.CabinetID and
			e.DataCenterID=f.DataCenterID AND b.Managed=true AND c.PrimaryIP>'' and c.SNMPFailureCount<3
			ORDER BY f.Name ASC, e.Location ASC;
```

**Reply by wilpig** [reply]
mainly i'm looking for the number of rows returned and that the deviceid that you are looking for is showing up in the result set.

**Reply by p1nn0chi0** [reply]
I'm sorry, not sure where to find the output of the query

**Reply by p1nn0chi0** [reply]
[pdu.txt](https://github.com/opendcim/openDCIM/files/13262594/pdu.txt)
I have attached the printout of the query.  5 rows, and DeviceID 57 does show in the query

**Reply by p1nn0chi0** [reply]
Each row shows a pdu device that I have put in the system so that checks out

**Comment by timlegge** [comment]
Hi

I have tested with the 8641 which is switched (each outlet can be
independently powered off.  Here is the configuration.  I don't believe the
8841 is switched so the outlet status will not be displayed.

Also, the device needs the "Poll" tag set (enter Poll and press enter then
save)

and under configuration on the reporting tab set the "Switches" to
either OptIn or OptOut

OptIn means any device polled for status updates upon viewing the device
must have a "Poll" tag.to pool the device
OptOut means any device polled for status updates (Switches PDUs) upon
viewing the device must have a "NoPoll" tag.to NOT poll the device

 [image: image.png]
Timothy Legge
***@***.***
***@***.***

**Reply by p1nn0chi0** [reply]
Thank you, I will try that out with the oids and see if I get better results

**Reply by alex001x** [reply]
OK, on the PDU APC, set to static IP, on SNMP -> Set V1 community "public"
on devices.php set OID
Check Manage = on
	                                       OID	                        Object	        Type	               Value
Firmware version OID	1.3.6.1.4.1.318.1.1.4.1.2.0	enterprises	OCTET-STRING	v1.2.0.4
Outlet factotry name OID				
Outlet user description OID				
Outlet count OID	         1.3.6.1.4.1.318.1.1.12.1.8.0	enterprises	INTEGER	42
Outlet State On/off				
Outlet state On	1			
Multiplier	10			
Phase 1	1.3.6.1.4.1.318.1.1.26.6.3.1.7.1		INTEGER	25
Phase 2	1.3.6.1.4.1.318.1.1.26.6.3.1.7.2		INTEGER	0
Phase 3	1.3.6.1.4.1.318.1.1.26.6.3.1.7.3		INTEGER	0
Processing Scheme	Combine3OIDWatts			
Voltage	240			
Amperage	32			

On device_templates.php
Set name :
IP 👍 
Select your device PDU APC
SNMP V1 comunity "public"
Save and test "set first outlet" and next

Good luck ;)

**Reply by alex001x** [reply]
can you send me your file cronjob ! thks

**Reply by alex001x** [reply]
Compare my OIDs with yours.

**Reply by p1nn0chi0** [reply]
<?php
  require( 'db.inc.php' );
  require( 'facilities.inc.php' );

  if ( php_sapi_name() != "cli" ) {
    echo "This script may only be run from the command line.";
    header( "Refresh: 5; url=" . redirect());
  }

  # Filter Types (Case Sensitive) - Filter Value
  # Country - 2 letter Country Code
  # Container - Numeric ContainerID
  # DataCenter - Numeric DataCenterID
  # Zone - Numeric ZoneID
  # Row - Numeric RowID
  # None - No filtering applied

  $filterType = "None";
  $filterValue = "";

  $PDU=new PowerDistribution( $filterType, $filterValue );

  $PDU->UpdateStats();

This is what I have under poll_pdu_stats.php

**Reply by p1nn0chi0** [reply]
<?php
  require( 'db.inc.php' );
  require( 'facilities.inc.php' );

  if ( php_sapi_name() != "cli" ) {
    echo "This script may only be run from the command line.";
    header( "Refresh: 5; url=" . redirect());
  }

  # Filter Types (Case Sensitive) - Filter Value
  # Country - 2 letter Country Code
  # Container - Numeric ContainerID
  # DataCenter - Numeric DataCenterID
  # Zone - Numeric ZoneID
  # Row - Numeric RowID
  # None - No filtering applied

  $filterType = "None";
  $filterValue = "";

  $PDU=new PowerDistribution( $filterType, $filterValue );

  $PDU->UpdateStats();

**Reply by p1nn0chi0** [reply]
Sorry....this is the file
<?php
  require( 'db.inc.php' );
  require( 'facilities.inc.php' );

  if ( php_sapi_name() != "cli" ) {
    echo "This script may only be run from the command line.";
    header( "Refresh: 5; url=" . redirect());
  }

  # Filter Types (Case Sensitive) - Filter Value
  # Country - 2 letter Country Code
  # Container - Numeric ContainerID
  # DataCenter - Numeric DataCenterID
  # Zone - Numeric ZoneID
  # Row - Numeric RowID
  # None - No filtering applied

  $filterType = "None";
  $filterValue = "";

  $PDU=new PowerDistribution( $filterType, $filterValue );

  $PDU->UpdateStats();

**Reply by p1nn0chi0** [reply]
<?php
  require( 'db.inc.php' );
  require( 'facilities.inc.php' );

  if ( php_sapi_name() != "cli" ) {
    echo "This script may only be run from the command line.";
    header( "Refresh: 5; url=" . redirect());
  }

  # Filter Types (Case Sensitive) - Filter Value
  # Country - 2 letter Country Code
  # Container - Numeric ContainerID
  # DataCenter - Numeric DataCenterID
  # Zone - Numeric ZoneID
  # Row - Numeric RowID
  # None - No filtering applied

  $filterType = "None";
  $filterValue = "";

  $PDU=new PowerDistribution( $filterType, $filterValue );

  $PDU->UpdateStats();

**Reply by alex001x** [reply]
YES  file poll_pdu_stats.php

**Reply by p1nn0chi0** [reply]
Also in the beginning of the response you said "set OID" in devices.php, what did you mean by that?

**Comment by p1nn0chi0** [comment]
[poll_pdu_stats.zip](https://github.com/opendcim/openDCIM/files/13070704/poll_pdu_stats.zip)

**Reply by p1nn0chi0** [reply]
Sorry, I'm a little new at this.  I have attached the file in zip format because github didn't like the .php format

**Comment by p1nn0chi0** [comment, marked as answer]
Still haven't gotten an answer so I will look at other products
